src/ui/public/timepicker/timepicker.js

```diff
@@ -139,17 +140,15 @@ module.directive('kbnTimepicker', function (quickRanges, timeUnits, refreshInter
       };
 
       $scope.applyRelative = function () {
-        $scope.from = getRelativeString();
-        $scope.to = 'now';
+        $scope.onFilterSelect(getRelativeString(), 'now');
       };
 
       function getRelativeString() {
         return 'now-' + $scope.relative.count + $scope.relative.unit + ($scope.relative.round ? '/' + $scope.relative.unit : '');
       }
 
       $scope.applyAbsolute = function () {
-        $scope.from = moment($scope.absolute.from);
-        $scope.to = moment($scope.absolute.to);
+        $scope.onFilterSelect($scope.absolute.from, $scope.absolute.to);
```

Comment:
It looks like the old code was cloning the absolute dates before setting them on from/to. I wonder if we should continue to clone them before passing to `onFilterSelect`? I checked the git logs but there's no hint why cloning my have been necessary. The current implementation seems to work fine, but I'd worry about introducing some subtle bug if we don't make this equivalent to the old code.